### PR TITLE
Add bitwise `xor` operation

### LINKED ISSUE #119: [feat] bitwise XOR not supported
Bitwise XOR (^) is not supported in Tact.

### LINKED ISSUE #235: Add `xor` support
`xor` should be added in stdlib.
```
@name(xor)
native xor(a: Int, b: Int): Int;
```
```
int xor(int x, int y) asm "XOR";
```

It's best to add it as an operator `^`.

### PR DESCRIPTION
**Pull request by Gusarich**
Closes #119

- [x] I have updated CHANGELOG.md
- [x] I have added unit tests to demonstrate the contribution is correctly implemented
- [x] I have run all the tests locally and no test failure was reported
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Display an error for integer overflow at compile-time: PR [#200](https://github.com/tact-lang/tact/pull/200)
 - Non-modifying `StringBuilder`'s `concat` method for chained string concatenations: PR [#217](https://github.com/tact-lang/tact/pull/217)
 - `toString` extension function for `Address` type: PR [#224](https://github.com/tact-lang/tact/pull/224)
+- `^` XOR operation: PR [#238](https://github.com/tact-lang/tact/pull/238)
```
Comment (anton-trunov) [inline CHANGELOG.md]:
```suggestion
- Bitwise XOR operation (`^`): PR [#238](https://github.com/tact-lang/tact/pull/238)
```

File: src/types/resolveConstantValue.ts [hunk 2]
```diff
@@ -29,6 +29,8 @@ function reduceIntImpl(ast: ASTExpression): bigint {
             return l & r;
         } else if (ast.op === "|") {
             return l | r;
+        } else if (ast.op === "^") {
+            return l ^ r;
```
Comment (anton-trunov) [inline src/types/resolveConstantValue.ts:33]: Let's create a `feature-const-eval` test file and add some tests for constant evaluation there. This most likely will work but we need to make sure we cover the case for negative numbers at least

File: src/test/feature-math.spec.ts [hunk 3]
```diff
@@ -42,6 +42,12 @@ describe("feature-math", () => {
         expect(await contract.getSub(1n, -2n)).toBe(3n);
         expect(await contract.getMul(2n, 2n)).toBe(4n);
         expect(await contract.getDiv(2n, 2n)).toBe(1n);
+        expect(await contract.getMod(2n, 2n)).toBe(0n);
+        expect(await contract.getShr(4n, 1n)).toBe(2n);
+        expect(await contract.getShl(2n, 1n)).toBe(4n);
+        expect(await contract.getAnd(2n, 3n)).toBe(2n);
+        expect(await contract.getOr(2n, 3n)).toBe(3n);
+        expect(await contract.getXor(2n, 3n)).toBe(1n);
```
Comment (anton-trunov) [inline src/test/feature-math.spec.ts:50]: let's cover also some cases of negative numbers

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Awesome!

File: src/types/resolveConstantValue.ts (hunk 2 above)
Comment (anton-trunov) [reply src/types/resolveConstantValue.ts:33]: Actually, we already have src/test/features/constants.tact, we just need to add some tests there

File: src/test/features/constants.tact [hunk 4]
```diff
@@ -46,6 +49,18 @@ contract ConstantTester {
...
+    }
+
+    get fun something10(): Int {
+        return self.something10;
+    }
+
+    get fun something11(): Int {
+        return self.something11;
+    }
+
```
Comment (anton-trunov) [inline src/test/features/constants.tact]: Looks like we are missing tests for this new part

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Almost ready for merging. We need to add tests for the new getters in `contstants.tact` and fix merge conflicts with the `main` branch

File: src/test/features/constants.tact (hunk 4 above)
Comment (Gusarich) [reply src/test/features/constants.tact]: these aren't tested in this version of `main`, as `feature-constants.spec.ts` was added in #237. I will rebase and update this PR now.

File: src/grammar/grammar.ohm [hunk 5]
```diff
@@ -143,6 +143,7 @@ Tact {
                     | ExpressionBinary "<<" ExpressionAdd --shl
                     | ExpressionBinary "&" ExpressionAdd --bin_and
                     | ExpressionBinary "|" ExpressionAdd --bin_or
+                    | ExpressionBinary "^" ExpressionAdd --bin_xor
```
Comment (anton-trunov) [inline src/grammar/grammar.ohm]: I believe there is a bug in this grammar: `^` has the same precedence as `&` and `^`. Incidentally, there is a bug with the rest of bitwise operators: e.g. `&` and `|` have the same precedence, and the shift operations too. See issue #263

**Review by anton-trunov — CHANGES_REQUESTED** [review]
There is an operator precedence issue, see #263. Let's fix the issue first and then rebase this PR on top of the fix. A separate fix is needed to not mix new tests with the ones in this PR.

File: src/test/features/math.tact
Comment (anton-trunov) [inline src/test/features/math.tact:1]: let's add a test with `^`, `&` and `|` to test their relative precedence

**Review by anton-trunov — APPROVED** [review]
great job!

**Pull request merged**